txnbuild/transaction.go

```diff
@@ -26,197 +27,645 @@ import (
 	"github.com/stellar/go/xdr"
 )
 
+// MinBaseFee is the minimum transaction fee for the Stellar network.
+const MinBaseFee = 100
+
 // Account represents the aspects of a Stellar account necessary to construct transactions. See
 // https://www.stellar.org/developers/guides/concepts/accounts.html
 type Account interface {
 	GetAccountID() string
-	IncrementSequenceNumber() (xdr.SequenceNumber, error)
-	// Action needed in release: horizonclient-v2.0.0
-	// add GetSequenceNumber method
-	// GetSequenceNumber() (xdr.SequenceNumber, error)
+	IncrementSequenceNumber() (int64, error)
+	GetSequenceNumber() (int64, error)
 }
 
-// Transaction represents a Stellar transaction. See
-// https://www.stellar.org/developers/guides/concepts/transactions.html
-type Transaction struct {
-	SourceAccount  Account
-	Operations     []Operation
-	BaseFee        uint32
-	Memo           Memo
-	Timebounds     Timebounds
-	Network        string
-	xdrTransaction xdr.TransactionV0
-	xdrEnvelope    *xdr.TransactionV0Envelope
+func hashHex(e xdr.TransactionEnvelope, networkStr string) (string, error) {
+	h, err := network.HashTransactionInEnvelope(e, networkStr)
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(h[:]), nil
 }
 
-// Hash provides a signable object representing the Transaction on the specified network.
-func (tx *Transaction) Hash() ([32]byte, error) {
-	return network.HashTransactionV0(tx.xdrTransaction, tx.Network)
+func concatSignatures(
+	e xdr.TransactionEnvelope,
+	networkStr string,
+	signatures []xdr.DecoratedSignature,
+	kps ...*keypair.Full,
+) ([]xdr.DecoratedSignature, error) {
+	// Hash the transaction
+	h, err := network.HashTransactionInEnvelope(e, networkStr)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to hash transaction")
+	}
+
+	extended := make(
+		[]xdr.DecoratedSignature,
+		len(signatures),
+		len(signatures)+len(kps),
+	)
+	copy(extended, signatures)
+	// Sign the hash
+	for _, kp := range kps {
+		sig, err := kp.SignDecorated(h[:])
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to sign transaction")
+		}
+		extended = append(extended, sig)
+	}
+	return extended, nil
 }
 
-// MarshalBinary returns the binary XDR representation of the transaction envelope.
-func (tx *Transaction) MarshalBinary() ([]byte, error) {
+func stringsToKP(
+	keys ...string,
+) ([]*keypair.Full, error) {
```

Comment:
This form of multiline formatting seems unnecessary here and elsewhere. When it is on one line it is still short. What's the goal with adopting that pattern here?
```suggestion
func stringsToKP(keys ...string) ([]*keypair.Full, error) {
```